tests/Serializer/AbstractSerializerTest.php

```diff
@@ -65,6 +65,34 @@ public function testObjectsAreNotStrings(): void
         $this->assertSame(['key' => 'value'], $result);
     }
 
+    public function traversableDataProvider(): \Generator
+    {
+        yield [new \ArrayIterator(['foo', 'bar']), ['foo', 'bar']];
+
+        // Also test with a non-rewindable non-cloneable iterator:
+        $newGenerator = static function (array $array): \Generator {
+            yield from $array;
+        };
+
+        yield [$newGenerator(['foo', 'bar']), ['foo', 'bar']];
```

Comment:
Can you please split each yielded value on its own line to improve readability? Also, you can inline the generator by wrapping the lambda with round brackets and yielding the values directly from inside it